docs/stake-registration.md

```diff
@@ -0,0 +1,147 @@
+
+The following subsections explain the different
+conditions presented in the diagram.
+
+#### Stake Validity
+
+A stake is valid and eligible for registration on the Babylon blockchain if
+it meets the following criteria:
+1. **Confirmed on the Bitcoin ledger**:
+   The staking transaction must have received sufficient
+   Bitcoin confirmations to be considered final.
+   The Babylon blockchain defines this in its
+   [module parameters](../proto/babylon/btccheckpoint/v1/params.proto)
+   (expected to be set to 10 confirmations for testnets and mainnet).
+2. **Compliance with Staking Parameters**:
+   Staking transactions must adhere to the Bitcoin block-range parameters
+   defined by the Babylon blockchain, which are consistent with those of
+   the Phase-1 system (e.g., covenant keys, and minimum and maximum stake requirements).
+   Valid stakes include:
+   1. **Active stakes**: As classified by the Phase-1 system.
+   2. **Overflow stakes**: Stakes that were valid but did not fit within Phase-1
+      staking caps will be accepted in phase-2.
+   3. **Stakes with malformed OP_RETURN**: Transactions rejected in Phase-1
+      due to malformed `OP_RETURN` outputs are still valid for the Babylon blockchain
+      if they meet all other criteria. Unlike Phase-1, the Babylon blockchain
+      does not require or check `OP_RETURN` outputs.
+
+#### Stake in the Allow-List
+
+To ensure a secure and controlled onboarding of existing stakes
+and the creation of new ones,
+the Babylon blockchain employs an initial stake ingestion policy.
+This is implemented via an allow-list that specifies which staking
+transactions can be registered by their transaction hashes.
+
+It has two important properties:
+* **Hardcoded Allow-List**: This list is embedded in the Babylon blockchain
+  codebase and is visible to all users.
+* **Automatic Expiration**: The allow-list functionality has a fixed expiration
+  Babylon chain block height to prevent abuse.
+
+The allow-list contents and its expiration Babylon chain block height
+will be chosen according to the network policies (e.g., test environments
+might have different policies than mainnet ones).
+
+#### Finality Provider Registration
+
+The Babylon blockchain only accepts stakes that delegate to finality
+providers registered on the chain. Finality providers from Phase-1
+must register anew in phase-2 before their delegators' stakes
+can register.
+
+If that is you, learn how to set up your Babylon blockchain
+finality provider stack and how to register
+**If you are a finality provider**,
+follow the 
+[registration guide](https://github.com/babylonlabs-io/finality-provider/blob/main/docs/finality-provider-operation.md)
+to set up your finality provider stack and register on the Babylon blockchain.
+
+## Stake Registration Process
+
+Registering an eligible stake involves submitting your Bitcoin staking
+transaction and supplementary data to the Babylon blockchain.
+Key data include:
```

Comment:
Maybe it is worth mentioning that the provided data must be valid according to the parameters which are active and staking tranasction inclusion height.